helm/external-dns-app/values.yaml

```diff
@@ -2,46 +2,175 @@
+  # cannot be changed.
+  zoneType:
+
+# externalDNS
+# Configuration options for external-dns itself.
+externalDNS:
+
+  # externalDNS.annotationFilter
+  # Annotation to filter objects to reconcile. Use in conjunction with
+  # `externalDNS.sources` to limit futher. This must be provided when running multiple
+  # instances of external-dns.
+  annotationFilter: "giantswarm.io/external-dns=managed"
+
+  # externalDNS.domainFilterList
+  # One or more domains that this instance of external-dns will manage. If
+  # no domains are provided then the cluster baseDomain must be provided via
+  # either `baseDomain` or `aws.baseDomain` (depending on the Route53 access
+  # method used).
+  domainFilterList: []
+  # - foo.bar.com
+  # - baz.bar.com
+
+  # externalDNS.interval
+  # Interval between synchronisations. Must be in duration format (e.g. `5m`)
+  interval:
+
+  # externalDNS.namespaceFilter
+  # The namespace to limit sources of endpoints to. If left blank, all
+  # namespaces will be used.
+  namespaceFilter: kube-system
+
+  # externalDNS.registry
+  # Configuration options relating to ownership of created records.
+  registry:
+
+    # externalDNS.registry.txtPrefix
+    # String prefix applied to TXT registry records. See
+    # https://github.com/kubernetes-sigs/external-dns/blob/master/docs/proposal/registry.md
+    txtPrefix:
+
+  # externalDNS.sources
+  # Resource types queried for endpoints.
+  sources:
   - service
   # Disable ingress source by default.
   # - ingress
 
-metricsPort: 7979
+global:
+
+  # global.image
+  # Configuration options for images.
+  image:
+
+    # global.image.name
+    name: giantswarm/external-dns
+
+    # global.image.registry
+    registry: quay.io
+
+    # global.image.tag
+    # When updating tag make sure to also keep appVersion in Chart.yaml in sync.
+    tag: v0.7.6
+
+  # global.metrics
+  # Metrics configuration options
+  metrics:
+
+    # global.metrics.port
+    # The port to serve metrics on.
+    port: 7979
+
+    # global.metrics.scrape
+    # Enable/disable scraping by Prometheus.
+    scrape: true
+
+  # global.resources
+  # Set pod resource limits and requests. This is passed as a YAML
+  # dictionary directly into the deployment manifest.
+  resources:
+    limits:
+      memory: 50Mi
+    requests:
+      cpu: 50m
+      memory: 50Mi
+
+  # global.securityContext
+  securityContext:
+
+    # global.securityContext.fsGroupID
+    fsGroupID: 65534
+
+    # global.securityContext.groupID
+    groupID: 65534
 
-image:
-  registry: quay.io
-  name: giantswarm/external-dns
-  # When updating tag make sure to also keep appVersion in Chart.yaml in sync
-  tag: v0.7.6
+    # global.securityContext.userID
+    userID: 65534
 
-resources:
-  limits:
-    memory: 50Mi
-  requests:
-    cpu: 50m
-    memory: 50Mi
+# provider
+# Identifies the cloudprovider. Currently supported: `aws` and `azure`. This *must*
+# be set.
+provider: changeme
```

Comment:
Can we make this property required via [json schema](https://json-schema.org/understanding-json-schema/reference/object.html#required-properties)?

Reply:
we can, however a more simple method might be to default to a null value in the values file and set the schema to be "string" only (don't allow "null") - that way the default values file will fail if the user doesn't provide it, but it will still work as a default app because this is always set via the cluster values.

Reply:
ok it is fine too